agent-antagonist/src/main.rs

```diff
@@ -0,0 +1,349 @@
+
+                        loop {
+                            info!(log, "requesting region {:?}", region_id);
+                            let client = get_client(&agent);
+                            let r = client.region_create(&region_request).await;
+                            drop(client);
+
+                            let region = match r {
+                                Ok(region) => {
+                                    info!(log, "requested region {:?} ok", region_id);
+                                    region
+                                }
+
+                                Err(e) => {
+                                    error!(log, "requesting region {:?} failed with {:?}", region_id, e);
+                                    break 'outer;
+                                }
+                            };
+
+                            match region.state {
+                                RegionState::Requested => {
+                                    info!(log, "waiting for region {:?}: state {:?}", region_id, region.state);
+                                    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+                                }
+
+                                RegionState::Created => {
+                                    info!(log, "region {:?} state {:?}", region_id, region.state);
+                                    break;
+                                }
+
+                                _ => {
+                                    info!(log, "region {:?} unknown state {:?}", region_id, region.state);
+                                    break 'outer;
+                                }
+                            }
+                        }
+
+                        // Optionally snapshot said region
+                        if random() {
+                            let snapshot_id = Uuid::new_v4();
+                            let snapshot = format!("{}/regions/{:?}@{:?}", dataset, region_id, snapshot_id);
+                            command(&log, "/usr/sbin/zfs", &["snapshot", &snapshot]).unwrap();
+
+                            let client = get_client(&agent);
+                            let r = client.region_get_snapshots(&RegionId(region_id.to_string())).await;
+                            drop(client);
+                            match r {
+                                Ok(_) => {
+                                    info!(log, "get region {:?} snapshots ok", region_id);
+                                }
+
+                                Err(e) => {
+                                    info!(log, "get region {:?} snapshots failed with {:?}", region_id, e);
+                                    break 'outer;
+                                }
+                            }
+
+                            let client = get_client(&agent);
+                            let r = client.region_get_snapshot(&RegionId(region_id.to_string()), &snapshot_id.to_string()).await;
```

Comment:
Did `cargo fmt` leave this line longer than 80 characters?  Is it longer than 80?

Reply:
It's definitely longer, I wonder why fmt didn't pick it up?